pkg/terminal/command.go

```diff
@@ -1898,55 +1899,106 @@ func (c *Commands) sourceCommand(t *Term, ctx callContext, args string) error {
 	return c.executeFile(t, args)
 }
 
-var disasmUsageError = errors.New("wrong number of arguments: disassemble [-a <start> <end>] [-l <locspec>]")
+func indexNextArgs(args string, splitStrs []string) int {
+	for i := range args {
+		for _, splitStr := range splitStrs {
+			if i+len(splitStr) <= len(args) && args[i:i+len(splitStr)] == splitStr {
+				return i
+			}
+		}
+	}
+	return -1
+}
 
-func disassCommand(t *Term, ctx callContext, args string) error {
-	var cmd, rest string
+func nextDisasmArgs(args string, splitStrs []string) (string, string, error) {
```

Comment:
I think we should use `split2PartsBySpace` instead of doing this.

Reply:
> I think we should use `split2PartsBySpace` instead of doing this.

Do this because I notice go allow this expression `main      .       main` for `-l`. 
so 
```
disasm   -l    main         .   main     -f    intel 
``` 
is ok.  

Even 
```
// main.go
package main

import (
	"fmt"
)

func main() {
	fmt   .       Printf("hello world\n")
}
```
`go run main.go` run ok.

Reply:
Yes but consider this:

```
(dlv) disasm -l m["a          string    -f      "]
```